check_docker.go

```diff
@@ -201,26 +214,36 @@ func (cd *CheckDocker) CheckImageContainerIsInGoodShape(imageId string) *nagios.
 	return &nagios.NagiosStatus{fmt.Sprintf("Container(ID: %v) of image: %v is in top shape.", containerRunning.ID, imageId), nagios.NAGIOS_OK}
 }
 
+func init() {
+	flag.StringVar(&endpoint, "base-url", "http://localhost:2375", "The Base URL for the Docker server")
+	flag.Float64Var(&warnMetaSpace, "warn-meta-space", 100, "Warning threshold for Metadata Space")
+	flag.Float64Var(&critMetaSpace, "crit-meta-space", 100, "Critical threshold for Metadata Space")
+	flag.Float64Var(&warnDataSpace, "warn-data-space", 100, "Warning threshold for Data Space")
+	flag.Float64Var(&critDataSpace, "crit-data-space", 100, "Critical threshold for Data Space")
+	flag.StringVar(&imageId, "image-id", "", "An image ID that must be running on the Docker server")
+	flag.StringVar(&tlsCertPath, "tls-cert", "", "Path to TLS cert file.")
+	flag.StringVar(&tlsKeyPath, "tls-key", "", "Path to TLS key file.")
+	flag.StringVar(&tlsCAPath, "tls-ca", "", "Path to TLS CA file.")
+}
+
 func main() {
-	dockerEndpoint := *flag.String("base-url", "http://localhost:2375", "The Base URL for the Docker server")
+	flag.Parse()
 
-	cd, err := NewCheckDocker(dockerEndpoint)
```

Comment:
this is called prematurely before `flag.Parse()` is called, hence `dockerEndpoint` still being a default value.